geonode/catalogue/views.py

```diff
@@ -330,3 +330,8 @@ def csw_render_extra_format_html(request, layeruuid, resname):
     extra_res_md['poc_email'] = pocp.email
     return render(request, "geonode_metadata_full.html", context={"resource": resource,
                                                                   "extra_res_md": extra_res_md})
+
+
+def resolve_uuid(request, uuid):
```

Comment:
Can we please add here some sanity checks on:

- [ ] Input parameters
- [ ] Resource existence on the DB
- [ ] Resource access permissions

Also please add some tests for this simple endpoint